### ISSUE TITLE
build image without docker

### ISSUE THREAD

**Issue by Arryboom** [kubernetes/kompose#1568]
as kompose is usually for running docker compose files in kubernetes,usually we dont has docker installed in kubernetes because now kubenetes not need docker by default anymore,could we support ```kompose convert --build=local ``` with other  options which is standalone image builder like 

- [Buildah](https://github.com/containers/buildah "https://github.com/containers/buildah") specializes in building [Open Container Initiative](https://opencontainers.org/) (OCI) images. Its commands replicate all of a Dockerfile’s commands. It enables building images with and without Dockerfiles, without requiring root privileges.
- [img](https://github.com/genuinetools/img "https://github.com/genuinetools/img") is a standalone, daemon-less, unprivileged Dockerfile- and OCI-compatible container image builder. It’s cache efficient and can execute multiple build stages concurrently, as it internally uses [BuildKit’s DAG solver](https://github.com/moby/buildkit "https://github.com/moby/buildkit") as its image builder.

as image builder?

as now I got 
```
[root@master1 Open-Assistant]# kompose convert --build=local -f docker-compose.yaml -v
DEBU Checking validation of provider: kubernetes
DEBU Checking validation of controller:
WARN Restart policy 'unless-stopped' in service inference-server is not supported, convert it to 'always'
WARN Restart policy 'unless-stopped' in service inference-text-client is not supported, convert it to 'always'
INFO Network open-assistant-default is detected at Source, shall be converted to equivalent NetworkPolicy at Destination
INFO Build key detected. Attempting to build image 'oasst-backend'
DEBU Build image working dir is: /root/Open-Assistant
DEBU Build image service build is: /root/Open-Assistant
DEBU Build image context is: /root/Open-Assistant
INFO Building image 'oasst-backend' from directory 'Open-Assistant'
DEBU Created temporary file /tmp/kompose-image-build-2767120133 for Docker image tarballing
DEBU Image oasst-backend build output:
FATA Unable to build Docker image for service backend: Unable to build image. For more output, use -v or --verbose when converting.: dial unix /var/run/docker.sock: connect: no such file or directory
[root@master1 Open-Assistant]# kompose convert --build=local -f docker-compose.yaml -vvv
DEBU Checking validation of provider: kubernetes
DEBU Checking validation of controller:
WARN Restart policy 'unless-stopped' in service inference-text-client is not supported, convert it to 'always'
WARN Restart policy 'unless-stopped' in service inference-server is not supported, convert it to 'always'
INFO Network open-assistant-default is detected at Source, shall be converted to equivalent NetworkPolicy at Destination
INFO Build key detected. Attempting to build image 'oasst-backend'
DEBU Build image working dir is: /root/Open-Assistant
DEBU Build image service build is: /root/Open-Assistant
DEBU Build image context is: /root/Open-Assistant
INFO Building image 'oasst-backend' from directory 'Open-Assistant'
DEBU Created temporary file /tmp/kompose-image-build-2158795954 for Docker image tarballing
DEBU Image oasst-backend build output:
FATA Unable to build Docker image for service backend: Unable to build image. For more output, use -v or --verbose when converting.: dial unix /var/run/docker.sock: connect: no such file or directory
```  
and dont really want to install docker only for the image build work,as we usually running kompose in kubernetes machine, it might will cause conflict with the components of kubernets

**Comment by AhmedGrati** [comment]
Hey @Arryboom, thank you for submitting this issue. I see that you have a valid point here, and it would be a great addition to support other tools to build container images other than Docker. WDYT @cdrage?
BTW @Arryboom, if you want to work on it feel free to tell me so that I can assign it to you. Otherwise, we'll tackle it asap. 🙏

**Comment by AhmedGrati** [comment]
Closed by #1624.
